### PR TITLE
feat(edgegw): add loadbalancer

### PR DESCRIPTION
**Pull request by gaetanars**
### Description of your changes
Add loadbalancer to the edgegateway + use superschema

If you submit change in the provider code, please make sure to:

- [x] Write or modify examples in `examples/` directory
- [x] Write or modify acceptance tests
- [x] Run `make generate` to ensure the doc was updated properly

### How has this code been tested
All acceptance tests on edgegw ar OK

### CODE REVIEW

File: docs/data-sources/edgegateway.md [hunk 1]
```diff
@@ -2,12 +2,12 @@
 page_title: "cloudavenue_edgegateway Data Source - cloudavenue"
 subcategory: "Edge Gateway (Tier-1)"
 description: |-
-  The edge gateway data source show the details of the edge gateway.
+  The Edge Gateway  data source allows you to show the details of an Edge Gateways in CloudAvenue.
```
Comment (dmicheneau) [inline docs/data-sources/edgegateway.md:5]: delete one space between Gateway and data.

File: docs/data-sources/edgegateway.md [hunk 2]
```diff
@@ -2,12 +2,12 @@
...
 ---
 
 # cloudavenue_edgegateway (Data Source)
 
-The edge gateway data source show the details of the edge gateway.
+The Edge Gateway  data source allows you to show the details of an Edge Gateways in CloudAvenue.
```
Comment (dmicheneau) [inline docs/data-sources/edgegateway.md:10]: delete one space between Gateway and data.

File: docs/data-sources/edgegateway.md [hunk 3]
```diff
@@ -30,9 +30,10 @@ output "gateway" {
 
 ### Read-Only
 
-- `description` (String) The description of the Edge Gateway.
+- `description` (String) The description of the owner of the Edge Gateway.
```
Comment (dmicheneau) [inline docs/data-sources/edgegateway.md:33]: try with: The description of the edge gateway owner.

File: docs/data-sources/edgegateway.md [hunk 4]
```diff
@@ -30,9 +30,10 @@ output "gateway" {
...
 - `id` (String) The ID of the Edge Gateway.
+- `lb_enabled` (Boolean) Load Balancing state on the Edge Gateway.
 - `owner_name` (String) The name of the owner of the Edge Gateway.
-- `owner_type` (String) The type of the owner of the Edge Gateway.
-- `tier0_vrf_id` (String) The ID of the Tier-0 VRF.
+- `owner_type` (String) The type of the owner of the Edge Gateway (vdc|vdc-group).
```
Comment (dmicheneau) [inline docs/data-sources/edgegateway.md:37]: try with: The description of the edge gateway owner.

File: internal/provider/edgegw/edgegateway_resource.go [hunk 5]
```diff
@@ -457,6 +391,80 @@ func (r *edgeGatewaysResource) Update(
...
+		return nil
+	})
+
+	if errRetry != nil {
+		resp.Diagnostics.AddError("Error waiting job to complete", errRetry.Error())
+		return
+	}
+
+	// Write logs using the tflog package
+	tflog.Trace(ctx, "Edge Gateway updated")
```
Comment (azrod) [inline internal/provider/edgegw/edgegateway_resource.go:464]: Realy necessary ?

File: internal/provider/edgegw/edgegateway_schema.go [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+	return superschema.Schema{
+		Common: superschema.SchemaDetails{
+			MarkdownDescription: "The Edge Gateway ",
+		},
+		Resource: superschema.SchemaDetails{
+			MarkdownDescription: " resource allows you to create and delete Edge Gateways in CloudAvenue.",
+		},
+		DataSource: superschema.SchemaDetails{
+			MarkdownDescription: " data source allows you to show the details of an Edge Gateways in CloudAvenue.",
+		},
```
Comment (azrod) [inline internal/provider/edgegw/edgegateway_schema.go:56]:
```suggestion
		Resource: superschema.SchemaDetails{
			MarkdownDescription: " resource allows you to create and delete Edge Gateways in Cloud Avenue.",
		},
		DataSource: superschema.SchemaDetails{
			MarkdownDescription: " data source allows you to show the details of an Edge Gateways in Cloud Avenue.",
		},
```

File: internal/provider/edgegw/edgegateway_schema.go [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+				},
+				Resource: &schemaR.StringAttribute{
+					PlanModifiers: []planmodifier.String{
+						stringplanmodifier.UseStateForUnknown(),
+					},
+				},
+				DataSource: &schemaD.StringAttribute{
+					Computed: true,
+				},
+			},
```
Comment (azrod) [inline internal/provider/edgegw/edgegateway_schema.go:79]: If value is computed is common is not necessary to specify in DataSource

File: docs/resources/edgegateway.md [hunk 8]
```diff
@@ -2,12 +2,12 @@
 page_title: "cloudavenue_edgegateway Resource - cloudavenue"
 subcategory: "Edge Gateway (Tier-1)"
 description: |-
-  The Edge Gateway resource allows you to create and delete Edge Gateways in CloudAvenue.
+  The Edge Gateway  resource allows you to create and delete Edge Gateways in CloudAvenue.
```
Comment (dmicheneau) [inline docs/resources/edgegateway.md:5]: delete one space between Gateway and resource.

File: docs/resources/edgegateway.md [hunk 9]
```diff
@@ -2,12 +2,12 @@
...
 ---
 
 # cloudavenue_edgegateway (Resource)
 
-The Edge Gateway resource allows you to create and delete Edge Gateways in CloudAvenue.
+The Edge Gateway  resource allows you to create and delete Edge Gateways in CloudAvenue.
```
Comment (dmicheneau) [inline docs/resources/edgegateway.md:10]: delete one space between Gateway and resource.

File: internal/tests/edgegw/edgegateway_resource_test.go [hunk 10]
```diff
@@ -55,6 +56,7 @@ func TestAccEdgeGatewayResource(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "owner_name", "MyVDC"),
 					resource.TestCheckResourceAttr(resourceName, "tier0_vrf_name", "prvrf01eocb0006205allsp01"),
```
Comment (dmicheneau) [inline internal/tests/edgegw/edgegateway_resource_test.go:57]: i'm not convince by the way to test the name of a datasource in this resource.

**Review by azrod — APPROVED** [review]

**Pull request merged**